cabra/src/pages/components/Task.tsx

```diff
@@ -1,20 +1,55 @@
 import "twin.macro";
 
+import { getTask, getTasks, updateTask } from "../../api/tasks";
+import { useMutation, useQueryClient } from "react-query";
+
+import CheckBox from "./CheckBox";
 import { ITask } from "../../types/task";
 import { Link } from "react-router-dom";
 import NavigationButton from "./NavigationButton";
 import { formatDateTime } from "../../utils/dates";
 import { routeHelpers } from "../../routes";
+import { useState } from "react";
 
 interface Props {
   task: ITask;
   detailsLink: boolean;
 }
 
 export default function Task({ task, detailsLink }: Props) {
+  const [errorMessage, setErrorMessage] = useState("");
+  const client = useQueryClient();
+  const addTask = useMutation((task: ITask) => updateTask(task.id, task), {
+    onSuccess: () => {
+      client.invalidateQueries(getTasks.cacheKey);
+      client.invalidateQueries(getTask.cacheKey(task.id));
+    },
+  });
+
   return (
     <div tw="rounded-xl w-full bg-white shadow-2xl text-blue-800  p-4">
-      <p tw="text-xl md:text-2xl">{task.title}</p>
+      <p tw="flex items-center text-xl md:text-2xl">
+        {task.title}
+        {
+          <CheckBox
+            checked={task.isDone}
+            onChange={async (event) => {
+              try {
+                const updatedTask = {
+                  ...task,
+                  isDone: !task.isDone,
+                };
+                await addTask.mutateAsync(updatedTask);
+                event.target.checked;
+              } catch (error) {
+                setErrorMessage(
+                  "Something went wrong :C, It's not possible to uncheck the task."
+                );
+              }
+            }}
+          />
```

Comment:
Long functions like this one make the UI code harder to read so it's a bit nicer to have them defined before the JSX and just pass them by reference:
```jsx
const handleIsDoneChange = async (event) => {....};

// and then in JSX

<CheckBox 
  checked={task.isDone}
  onChange={handleIsDoneChange}
/>
```